pkg/rpc/client/rpc.go

```diff
@@ -588,3 +598,42 @@ func (c *Client) CalculateValidUntilBlock() (uint32, error) {
 	}
 	return blockCount + validatorsCount, nil
 }
+
+// AddNetworkFee adds network fee for each witness script to transaction.
+func (c *Client) AddNetworkFee(tx *transaction.Transaction, acc *wallet.Account) error {
+	size := io.GetVarSize(tx)
+	if acc.Contract != nil {
+		tx.NetworkFee += calculateNetworkFee(acc.Contract.Script, &size)
+	}
+	for _, cosigner := range tx.Cosigners {
+		contract, err := c.GetContractState(cosigner.Account)
+		if err != nil {
+			return err
+		}
+		tx.NetworkFee += calculateNetworkFee(contract.Script, &size)
+	}
+	//TODO:
+	//tx.NetworkFee += size * NativeContract.Policy.GetFeePerByte;
+	return nil
+}
+
+func calculateNetworkFee(script []byte, size *int) util.Fixed8 {
+	var netFee util.Fixed8
+	if vm.IsSignatureContract(script) {
+		*size += 67 + io.GetVarSize(script)
```

Comment:
Magic constant.

Reply:
Actually, I don't quite understand what are the corresponding fields for these constants and some opcodes, but it works like a magic (checked on real transactions in `TestCreateBasicChain` method)